joblib/parallel.py

```diff
@@ -211,6 +213,19 @@ def _get_active_backend(
         thread_config['n_jobs'] = 1
         return sharedmem_backend, thread_config
 
+    force_processes = (
+        not explicit_backend and prefer == 'processes' and uses_threads
+    )
```

Comment:
Ok make sense compare to the fuzzy logic that was implemented in #1623 
I would maybe move this before line 197, to have all the variables in one place and check that there is no conflict?